# 1. Introduction

# 2. Import and display the dataset

In [15]:
import seaborn as sns # import seaborn package
tips = sns.load_dataset("tips") # import tips dataset from seaborn
tips.head(10) # show first 10 rows of tips dataset

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4
5       25.29  4.71    Male     No  Sun  Dinner     4
6        8.77  2.00    Male     No  Sun  Dinner     2
7       26.88  3.12    Male     No  Sun  Dinner     4
8       15.04  1.96    Male     No  Sun  Dinner     2
9       14.78  3.23    Male     No  Sun  Dinner     2

Printing the first 10 rows of the dataset gives a feel for the data involved. We can see that, for each party, we know the total bill, tip given, size of party, day and time of reservation. There is also a column for "sex" one for "smoker" in the dataset.

# 3. Descriptive Statistics

Some summary statistics about the dataset are shown below:

In [3]:
tips.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

We can see that the values for the total bill range from $3.07 to $50.81 with a mean bill of just under $20 and a median bill of just under $18. The values for tips range from $1 to $10 with an average tip of just under $3 and very similar median value. The mean value for size is about 2.5; the smallest party was 1 and the largest was a party of 6.

# 4. Analysis of Relationship between Bill and Tip

# 5. Analysis of Relationship between Variables

# 6. Conclusion